torch/_subclasses/fake_tensor.py

```diff
@@ -99,13 +105,24 @@ def _is_tensor_constructor(func: OpOverload):
     )
 
 
+@functools.lru_cache(None)
+def get_schema_info(func):
+    return torch._C._SchemaInfo(func._schema)  # type: ignore[attr-defined]
+
+
+def tree_flatten_only(ty: Type[T], pytree: PyTree):
+    flat_vals, _ = tree_flatten(pytree)
+    return [elem for elem in flat_vals if isinstance(elem, ty)]
```

Comment:
move this to pytree

Reply:
will do on follow up

Reply:
`tree_flatten` returns both the output and the spec, but it's not clear to me how to return the spec with `tree_flatten_only` and i think it would be a misleading api if `tree_flatten` returned the pytree but `tree_flatten_only` did not... Thoughts ?

Reply:
When slots are "not selected", they should be incorporated into the spec. You may have to redo your implementation so that this works